ql/src/test/org/apache/hadoop/hive/ql/parse/repl/metric/TestReplicationMetricCollector.java

```diff
@@ -75,6 +84,12 @@ public void setup() throws Exception {
     MetricCollector.getInstance().init(conf);
     Mockito.when(fmd.getFailoverEventId()).thenReturn(10L);
     Mockito.when(fmd.getFilePath()).thenReturn("dummyDir");
+    disableBackgroundThreads();
+  }
+
+  private void disableBackgroundThreads() {
+    PowerMockito.mockStatic(MetricSink.class);
+    Mockito.when(MetricSink.getInstance()).thenReturn(metricSinkInstance);
```

Comment:
I'm new to the this testing framework - but I assume my understanding is correct:
This works because metricSinkInstance gets its MetricSink() constructor called because it is annotated with Mock and since the metricSinkInstance never gets init called on it, it never starts the executorService, correct?

Reply:
Yes, this is the main idea and I verified locally with the debugger that the background threads are not started. Precisely what happens is that whenever MetricSink.getInstance() is called, it returns the mock metricSinkInstance and even if its init() is called, it does nothing because it's an empty method provided by the mock framework.